rust/src/pgsql/pgsql.rs

```diff
@@ -341,6 +341,7 @@ impl PgsqlState {
             );
             match PgsqlState::state_based_req_parsing(self.state_progress, start) {
                 Ok((rem, request)) => {
+                    sc_app_layer_parser_trigger_raw_stream_reassembly(flow, Direction::ToServer as i32);
```

Comment:
What is happening here ?
Is this the place for every pgsql tx creation ? Should this happen also for every protocol, not only pgsql ?

Reply:
asking you @jufajardini ;-)

Reply:
> asking you @jufajardini ;-)

Thankfully Victor and I talked about it this morning :P

> What is happening here ?

We tell the detection engine that this is ready to be inspected, no need to keep buffering more stream data before doing so.

 > Should this happen also for every protocol, not only pgsql ?
 
Indeed, should be done for every TCP app-layer proto which still doesn't have it. That's tracked with https://redmine.openinfosecfoundation.org/issues/7026 and will require crafting appropriate SV tests for each of those.

> Is this the place for every pgsql tx creation ?

So, in this case, we want to actually move this callback to a few lines further down, when we check whether the pgsql tx is completed or not.

Reply:
Thanks.

So, I wonder if there is a good generic place to do it instead of doing it in each app-layer proto.
There are 3 tx iterators (detection, logging and cleaning) that check for tx progress...
But maybe it is too late already at that point...

Reply:
> Thanks.
> 
> So, I wonder if there is a good generic place to do it instead of doing it in each app-layer proto. There are 3 tx iterators (detection, logging and cleaning) that check for tx progress... But maybe it is too late already at that point...

A long time to answer, but will do so, still, also as an exercise in understanding: I think that because: what means to have a tx completed varies per app-layer and who has that information at parsing time is the app-layer, itmakes sense to have it triggered by the app-layer proto - sort of like with frames. Does that sound correct?

Reply:
> what means to have a tx completed varies per app-layer and who has that information at parsing time is the app-layer, itmakes sense to have it triggered by the app-layer proto - sort of like with frames. Does that sound correct?

Yes, there may be no good way to centralize it...

For testing, you should have a pcap with a TCP packet that contains 2 Transactions/PDUs (IIRC there is some mqtt frames test like that) to check what happens in that case